### ISSUE TITLE
[Documentation] [Small] Change the tutorial data type for train_X and train_Y to doubles to avoid error message

### ISSUE THREAD

**Issue by vmallela0** [pytorch/botorch#1887]
## Issue description

When running the starter code in the README.md 'getting started" section, SingleTaskGP threw an error because the tensor dtype were floats instead of doubles

## Code example
`botorch/botorch/models/gpytorch.py:129: UserWarning: The model inputs are of type torch.float32. It is strongly recommended to use double precision in BoTorch, as this improves both precision and stability and can help avoid numerical errors. See https://github.com/pytorch/botorch/discussions/1444
  warnings.warn(_get_single_precision_warning(X.dtype), UserWarning)`

I would like to make a PR changing the "Fit a Gaussian Process model to data" section of getting started in Readme to 

~~~
import torch
from botorch.models import SingleTaskGP
from botorch.fit import fit_gpytorch_mll
from gpytorch.mlls import ExactMarginalLogLikelihood

train_X = torch.rand(10, 2)
train_X = train_X.to(torch.double)
Y = 1 - (train_X - 0.5).norm(dim=-1, keepdim=True)  # explicit output dimension
Y += 0.1 * torch.rand_like(Y)
train_Y = (Y - Y.mean()) / Y.std()
train_Y = train_Y.to(torch.double)

gp = SingleTaskGP(train_X, train_Y)
mll = ExactMarginalLogLikelihood(gp.likelihood, gp)
fit_gpytorch_mll(mll)
~~~

## System Info
Please provide information about your setup, including
- BoTorch Version: 0.8.6.dev39+g185e98fc
- GPyTorch Version: 1.10
- PyTorch Version: 2.0.1
- MacOS Ventura 13.4

**Comment by saitcakmak** [comment]
Hi @vmallela0. Thanks for flagging this. Your suggestion makes a lot of sense, and we'd be happy to accept a PR fixing it. We do not want code in the documentation to throw a bunch of warnings :). 

One small suggestion. Instead of 
```
train_X = torch.rand(10, 2)
train_X = train_X.to(torch.double)
```
you can just do
```
train_X = torch.rand(10, 2, dtype=torch.double)
```
which is cleaner and more efficient as it avoids having to re-allocate the tensor. Since `Y` & `train_Y` are derived from `train_X`, those should automatically be double precision, so you don't need this bit either.
```
train_Y = train_Y.to(torch.double)
```

**Comment by vmallela0** [comment]
@saitcakmak thanks for the tip :) made the pr #1888

**Comment by Balandat** [comment]
Another thing we could consider doing is to add `torch.set_default_dtype(torch.double)` at the header of the example with a comment that this is generally recommended when working with GP models (if mixing with NNs e.g. as in DKL then it may make sense to do things in mixed precision). It may take a bit of comment space to explain that though. @saitcakmak, @esantorella  wdyt?
